smart-contracts/contracts/cl-vault/src/vault/range.rs

```diff
@@ -258,54 +248,59 @@ pub fn do_swap_deposit_merge(
         },
     )?;
 
-    // start swap math
+    // start swap workflow
+
     // get remaining balance in contract for each token (one of these should be zero i think)
-    let balance0 =
-        querier.query_balance(env.contract.address.clone(), pool_config.token0.clone())?;
-    let balance1 =
-        querier.query_balance(env.contract.address.clone(), pool_config.token1.clone())?;
+    // @notice: this actually works if this function (do_swap_deposit_merge) is called by
+    // handle_initial_create_position_reply, double check if implementing it somewhere else
+    let (balance0, balance1) = CURRENT_REMAINDERS.load(storage)?;
```

Comment:
This might be missing funds from the 1st create position From create position Response